src/core/utils/useNotificationMessage.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { useState } from "react";
+import PromoBar from "../../components/promo-bar/PromoBar";
+
+type UseNotificationOptionsType = {
+  timeout?: number;
+  scrollToTop?: boolean;
+};
+type UseNotificationReturnType = [React.FC, (text: string) => void];
+
+export const useNotificationMessage = ({
+  timeout = 5000,
+  scrollToTop = true
+}: UseNotificationOptionsType = {}): UseNotificationReturnType => {
+  const [notificationMessage, setNotificationMessage] = useState<string | null>(
+    null
+  );
+
+  const handleNotificationMessage = (text: string) => {
+    setNotificationMessage(text);
+    if (scrollToTop) {
+      window.scrollTo(0, 0);
+    }
+    if (timeout) {
+      setTimeout(() => {
+        setNotificationMessage(null);
+      }, timeout);
+    }
+  };
+
+  const NotificationComponent: React.FC = () => {
+    if (!notificationMessage) return null;
+    return <PromoBar text={notificationMessage} type="info" />;
+  };
+
+  return [NotificationComponent, handleNotificationMessage];
```

Comment:
I think it is nice to have components isolated in separate files. That also makes it easier for possible unittests/snapshots.
Also it should be possible to make the message a prop for NotificationComponent to make it pure and, again, easier to test :):
```typescript
interface NotificationComponentProps {
  message: string;
};

const const NotificationComponent: React.FC<NotificationComponentProps> = ({message}) => <PromoBar text={message} type="info" />;
```

And then here:
```typescript
  return [(<NotificationComponent message={notificationMessage} />), handleNotificationMessage];

Reply:
Did not test it 😄